lightningd/payalgo.c

```diff
@@ -15,6 +16,107 @@
 #include <lightningd/subd.h>
 #include <sodium/randombytes.h>
 
+/* Record of failures. */
+enum pay_failure_type {
+	FAIL_UNPARSEABLE_ONION,
+	FAIL_PAYMENT_REPLY
+};
+/* A payment failure. */
+struct pay_failure {
+	/* Part of a list of failures */
```

Comment:
Generally prefer this comment to refer to the parent type, i.e. "Off struct pay_command pay_failures".

Reply:
okay.